lib/dor/models/publishable.rb

```diff
@@ -10,6 +10,58 @@ module Publishable
+       thumb_image=nil
+
+       # these are the finders we will use to search for a thumb resource in contentMetadata, they will be searched in the order provided, stopping when one is reached
+       thumb_xpath_finders = [
+           {image_type: 'local', finder: "/contentMetadata/resource[@type='thumb' and @thumb='yes']/file[#{mime_type_finder}]"},      # first find a file of mimetype jp2 explicitly marked as a thumb in the resource type and with a thumb=yes attribute
+           {image_type: 'external', finder: "/contentMetadata/resource[@type='thumb' and @thumb='yes']/externalFile[#{mime_type_finder}]"}, # same thing for external files
+           {image_type: 'local', finder: "/contentMetadata/resource[(@type='page' or @type='image') and @thumb='yes']/file[#{mime_type_finder}]"},# next find any image or page resource types with the thumb=yes attribute of mimetype jp2
+           {image_type: 'external', finder: "/contentMetadata/resource[(@type='page' or @type='image') and @thumb='yes']/externalFile[#{mime_type_finder}]"},# same thing for external file
+           {image_type: 'local', finder: "/contentMetadata/resource[@type='thumb']/file[#{mime_type_finder}]"}, # next find a file of mimetype jp2 or resource type=thumb but no marked with the thumb directive
+           {image_type: 'external', finder: "/contentMetadata/resource[@type='thumb']/externalFile[#{mime_type_finder}]"}, # same thing for external file
+           {image_type: 'local', finder: "/contentMetadata/resource[@type='page' or @type='image']/file[#{mime_type_finder}]"}, # finally find the first very first page or image resource of mimetype jp2
+           {image_type: 'external', finder: "/contentMetadata/resource[@type='page' or @type='image']/externalFile[#{mime_type_finder}]"} # same thing for external file
+         ]
+              
+       thumb_xpath_finders.each do |search_path|
+         thumb_files = cm.xpath(search_path[:finder]) # look for a thumb  
+         if thumb_files.size > 0   # if we find one, return the filename based on whether it is a local file or external file
+           if search_path[:image_type] == 'local'
+             thumb_image="#{remove_druid_prefix}/#{thumb_files[0]['id']}"
+           else
+             thumb_image="#{remove_druid_prefix(thumb_files[0]['objectId'])}/#{thumb_files[0]['fileId']}"
+           end
+           break  # break out of the loop so we stop searching
+         end
+       end 
+              
+       thumb_image
+    end
+
+    # Return a URI encoded version of the thumb image for use by indexers (leaving the extension of the filename)
+    # @return [String] URI encoded version of the thumb with the druid prefix, e.g. oo000oo0001%2Ffilenamewith%20space.jp2
+    def encoded_thumb
+      return unless thumb
+      thumb_druid=thumb.split('/').first # the druid (before the first slash)
+      thumb_filename=thumb.split(/[a-zA-Z]{2}[0-9]{3}[a-zA-Z]{2}[0-9]{4}[\/]/).last # everything after the druid
```

Comment:
agreed it would be nice to centralize, and could be easy as a something like 

```def pid_regex 
  "[a-zA-Z]{2}[0-9]{3}[a-zA-Z]{2}[0-9]{4}"
end```

and then substitution that in as needed.

But where to put it?  Are there any modules that are available everywhere?  Should there be one?  Same concept applies to the "remove_druid_prefix" method...it is applicable broadly but it has to go somewhere and I want to be sure it is always available.  At first I put the thumb stuff in "contentable", but that isn't available to the Dor::Collection...which surprisingly, can have a thumb and can have contentMetadata...